app/cdap/components/shared/ConfigurationGroup/index.tsx

```diff
@@ -153,7 +153,7 @@ const ConfigurationGroupView: React.FC<IConfigurationGroupProps> = ({
     let isUpdated = false;
     configGroup.map((config) => {
       config.properties.map((property) => {
-        if (property.show && newValues[property.name] == null) {
+        if (property.show && newValues[property.name] == null && property.defaultValue != null) {
```

Comment:
`===` and `!==` ?

Reply:
The value can be undefined or null, this way both can be true

Reply:
Then we could replace with `!newValues[property.name]`?
We get `if (property.show && !newValues[property.name] && property.defaultValue)`
I'd recommend not using `==` and `!=`, sometimes it introduces implicit bugs